source/server/admin/stats_params.h

```diff
@@ -32,6 +35,7 @@ struct StatsParams {
   StatsFormat format_{StatsFormat::Text};
   std::string filter_string_;
   absl::optional<std::regex> filter_;
+  std::shared_ptr<re2::RE2> safe_filter_;
```

Comment:
After https://github.com/envoyproxy/envoy/pull/19473 merged, we have two matcher engines and the benchmark result shows that it is faster than RE2. Although it belongs to `contrib`, for expansion considerations, could we use more flexible way to use the matcher engine?

/cc @zhxie

Reply:
3 matcher engines. The default for this endpoint is std::regex.

I don't want to add a dependency from core into contib. Do you have a specific suggestion here?

Reply:
Oh, I miss it, I don't want to add it either. 

@zhxie is it possible to adapt different matcher classes here rather than introduce dependencies?

Reply:
@daixiang0 no, they are in different scope.

Reply:
And can we use `Envoy::Regex::CompiledGoogleReMatcher` instead of `re2::RE2`?

Reply:
Good question: maybe it's worth it, but the test I'm making here that's harmonious with the existing filter behavior is PartialMatch, and that one calls FullMatch, so I'd have to add a new API. Not sure if that's worth it.

Also CompiledGoogleReMatcher doesn't have the ok() check -- I think from a UX perspective it would be better to return a 400 for an admin /stats?filter=bogus_regex request than simply have the match return nothing.